src/components/forms/AddressInput/index.tsx

```diff
@@ -39,50 +41,65 @@ const AddressInput = ({
   validators = [],
   defaultValue,
   disabled,
-}: AddressInputProps): React.ReactElement => (
-  <>
-    <Field
-      className={className}
-      component={TextField as any}
-      defaultValue={defaultValue}
-      disabled={disabled}
-      inputAdornment={inputAdornment}
-      name={name}
-      placeholder={placeholder}
-      testId={testId}
-      text={text}
-      type="text"
-      spellCheck={false}
-      validate={composeValidators(required, mustBeEthereumAddress, ...validators)}
-    />
-    <OnChange name={name}>
-      {async (value: string) => {
-        const address = trimSpaces(value)
-        // A crypto domain name
-        if (isValidEnsName(address) || isValidCryptoDomainName(address)) {
-          try {
-            const resolverAddr = await getAddressFromDomain(address)
-            const formattedAddress = checksumAddress(resolverAddr)
-            fieldMutator(formattedAddress)
-          } catch (err) {
-            logError(Errors._101, err.message)
-          }
-        } else {
-          // A regular address hash
-          let checkedAddress = address
-          // Automatically checksum valid (either already checksummed, or lowercase addresses)
-          if (isValidAddress(address)) {
+}: AddressInputProps): ReactElement => {
+  const [isVerifying, setIsVerifying] = useState<boolean>(false)
+  const adornment = isVerifying
+    ? {
+        endAdornment: (
+          <InputAdornment position="end">
+            <CircularProgress size="16px" />
+          </InputAdornment>
+        ),
+      }
+    : inputAdornment
+  return (
+    <>
+      <Field
+        className={className}
+        component={TextField as any}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        readOnly={isVerifying}
+        inputAdornment={adornment}
+        name={name}
+        placeholder={placeholder}
+        testId={testId}
+        text={text}
+        type="text"
+        spellCheck={false}
+        validate={composeValidators(required, mustBeEthereumAddress, ...validators)}
+      />
+      <OnChange name={name}>
+        {async (value: string) => {
+          setIsVerifying(true)
```

Comment:
only resolving ens is asynchronous, so perhaps it's better to set it to `true` only when the address is an ENS domain